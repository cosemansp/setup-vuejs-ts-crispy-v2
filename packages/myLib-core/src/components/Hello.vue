<template>
  <div class="hello">
    <h1>{{ msg }}</h1>
  </div>
</template>

<script lang="ts">
// Optional separated ts file
// export { default } from './Hello';
import { Component, Vue, Prop } from 'vue-property-decorator';

@Component({})
export default class Hello extends Vue {
  @Prop() message: String;
  msg = 'Welcome';

  created() {
    console.log('created');
  }
}
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped>
h1,
h2 {
  font-weight: normal;
}
</style>
