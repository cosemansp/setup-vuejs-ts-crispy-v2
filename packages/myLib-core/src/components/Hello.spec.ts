import { shallow } from '@vue/test-utils';
import Hello from './Hello.vue';

describe('Hello Component', () => {
  it('should have a default "msg" set to "Welcome"', () => {
    const component = new Hello();
    expect(component.msg).toBe('Welcome');
  });

  it('default render', () => {
    const wrapper = shallow(Hello);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
