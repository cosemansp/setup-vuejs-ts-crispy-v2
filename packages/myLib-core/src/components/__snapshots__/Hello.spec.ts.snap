// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello Component default render 1`] = `
<div class="hello">
  <h1>Welcome</h1>
</div>
`;
